### ISSUE TITLE
Alternate method to get CPU freq on OSX

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#975]
Current implementation is sub optimal as it returns the nominal current CPU frequency.
Note to self - try this approach: 
http://stackoverflow.com/a/14355710/376587

**Comment by amanusk** [comment]
If there is any update on how to obtain correct CPU frequency, I can help testing/implementing this.

**Comment by giampaolo** [comment]
Nope, I haven't got back to this since then. For the record, the problem is CPU frequency never changes. Apparently all approaches suggested in https://stackoverflow.com/questions/9948987 have the same problem.

**Comment by amanusk** [comment]
Indeed, non of them work from my testing. 
`Intel power gadget` shows CPU frequency changes, so it could be a good reference. Unfortunately, there are no sources available.
Perhaps https://github.com/kozlek/HWSensors can also help finding the way to do it 

![image](https://user-images.githubusercontent.com/7280933/41932690-58aae6e6-798a-11e8-909b-55b652f56549.png)
